frontend/src/components/Dashboard/SideBar.js

```diff
@@ -0,0 +1,261 @@
+import $ from 'jquery';
+import { lighten } from '@material-ui/core/es/styles/colorManipulator';
+import { renderSize, renderDuration } from '../../util/worksheet_utils';
+
+const styles = ({ spacing, palette }) => {
+    const family =
+        '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"';
+    return {
+        box: { marginLeft: 8, marginRight: 8, marginTop: 8, marginBottom: 0 },
+        progressBox: {
+            display: 'flex',
+            alignItems: 'center',
+            width: '100%',
+            height: 0,
+            paddingRight: 4,
+            paddingTop: 8,
+        },
+        card: {
+            display: 'flex',
+            flexDirection: 'column',
+            padding: spacing(2),
+            height: '100%',
+            boxShadow: '0 2px 4px 0 rgba(138, 148, 159, 0.2)',
+            '& > *:nth-child(1)': {
+                marginRight: spacing(2),
+            },
+            '& > *:nth-child(2)': {
+                flex: 'auto',
+            },
+        },
+        placeholderBox: {
+            marginBottom: 800,
+        },
+        avatar: { marginLeft: 8, marginTop: 12, marginBottom: 8 },
+        heading: {
+            fontFamily: family,
+            fontSize: 16,
+            marginBottom: 0,
+        },
+        subheader: {
+            fontSize: 14,
+            fontFamily: 'Roboto',
+            fontStyle: 'normal',
+            fontWeight: 300,
+            letterSpacing: 0.1,
+            color: '#000000',
+            lineHeight: '150%',
+            marginBottom: 0,
+        },
+        affiliation: {
+            fontSize: 15,
+            paddingBottom: 8,
+            fontFamily: 'Roboto',
+            fontStyle: 'normal',
+            fontWeight: 'normal',
+            letterSpacing: 0.15,
+            color: '#000000',
+        },
+        value: {
+            fontSize: 14,
+            color: palette.grey[500],
+        },
+        name: {
+            fontFamily: 'Roboto',
+            fontStyle: 'normal',
+            fontWeight: 500,
+            fontSize: 20,
+            letterSpacing: 0.15,
+        },
+        progress: {
+            minWidth: '85%',
+            maxWidth: '85%',
+        },
+    };
+};
+
+const BorderLinearProgress = withStyles({
+    root: {
+        height: 10,
+        width: 600,
+        backgroundColor: lighten('#3577cb', 0.5),
+    },
+    bar: {
+        borderRadius: 20,
+        backgroundColor: '#3577cb',
+    },
+})(LinearProgress);
+
+class SideBar extends React.Component {
+    /** Constructor. */
+    constructor(props) {
+        super(props);
+        this.state = {
+            bundles: [],
+        };
+    }
+
+    componentDidMount() {
+        const { classes } = this.props;
+        // Fetch bundles' count in different states owned by the current user one by one
+        const states: String[] = [
```

Comment:
This is a great variable to declare in constants.js instead.

Reply:
Good advice! Done.